sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JacksonParser.scala

```diff
@@ -418,6 +426,19 @@ class JacksonParser(
       }
     }
 
+    // When the input schema is setting to `nullable = false`, make sure the field is not null.
+    var index = 0
+    while (badRecordException.isEmpty && !skipRow && index < schema.length) {
+      if (!schema(index).nullable && row.isNullAt(index)) {
+        throw new IllegalSchemaArgumentException(
+          s"the null value found when parsing non-nullable field ${schema(index).name}.")
+      }
+      if (!checkedIndexSet.contains(index)) {
+        skipRow = structFilters.skipRow(row, index)
```

Comment:
is this change related to the nullability issue?

Reply:
yes, when the input JSON string contains the missing value field, `JacksonParser` should be able to skip rows through this null value field.

Reply:
This was not done before your PR? The missing field is always filled with null value, and we can use it to filter rows. This doesn't seem to be related to this PR.

Reply:
For example:
```scala
  val missingFieldInput = """{"c1":1}"""
  val schema = StructType(Seq(
    StructField("c1", IntegerType, nullable = true),
    StructField("c2", IntegerType, nullable = true)
  ))

  val filters = Seq(IsNotNull("c2"))

  val options = new JSONOptions(Map.empty[String, String], "GMT", "")
  val parser = new JacksonParser(schema, options, false, filters)
  val createParser = CreateJacksonParser.string _
  val result = parser.parse(missingFieldInput, createParser, UTF8String.fromString)
  println(result)
```

before this pr:
```
List([1,null])
```

after this pr:
```
List()
```

Please let me know if I've missed out anything.

Reply:
This is not a nullability issue, but a performance improvement. Spark will evaluate the filter again and drop the rows. Can we exclude this part from this PR? We need to backport the nullability fix but not the performance improvement.

Reply:
updated.